csharp/GildedRose.cs

```diff
@@ -4,86 +4,67 @@ namespace csharp
-        {
-            this.Items = Items;
-        }
+        private const string AGEDBRIE = "Aged Brie"; // AGED_BRIE ??
+        private const string SULFURAS = "Sulfuras, Hand of Ragnaros";
+        private const string TAFKAL80ETC = "Backstage passes to a TAFKAL80ETC concert";
+        private IList<Item> Items;
+
+        public GildedRose(IList<Item> Items) => this.Items = Items;
 
         public void UpdateQuality()
         {
-            for (var i = 0; i < Items.Count; i++)
+            foreach (var item in Items)
             {
-                if (Items[i].Name != "Aged Brie" && Items[i].Name != "Backstage passes to a TAFKAL80ETC concert")
-                {
-                    if (Items[i].Quality > 0)
-                    {
-                        if (Items[i].Name != "Sulfuras, Hand of Ragnaros")
-                        {
-                            Items[i].Quality = Items[i].Quality - 1;
-                        }
-                    }
-                }
-                else
-                {
-                    if (Items[i].Quality < 50)
-                    {
-                        Items[i].Quality = Items[i].Quality + 1;
-
-                        if (Items[i].Name == "Backstage passes to a TAFKAL80ETC concert")
-                        {
-                            if (Items[i].SellIn < 11)
-                            {
-                                if (Items[i].Quality < 50)
-                                {
-                                    Items[i].Quality = Items[i].Quality + 1;
-                                }
-                            }
+                bool isNotAgedbrie = item.Name != AGEDBRIE; // Very very great
+                bool isNotSulfuras = item.Name != SULFURAS; // GOOD good
 
-                            if (Items[i].SellIn < 6)
-                            {
-                                if (Items[i].Quality < 50)
-                                {
-                                    Items[i].Quality = Items[i].Quality + 1;
-                                }
-                            }
-                        }
-                    }
+                if (isNotAgedbrie)
+                {
+                    QualityLessThan50(item);
                 }
+                SulfurasQualityCase(item);
 
-                if (Items[i].Name != "Sulfuras, Hand of Ragnaros")
+                if (isNotSulfuras)
                 {
-                    Items[i].SellIn = Items[i].SellIn - 1;
+                    item.SellIn -= 1; // Ajouter une fonction qui update la qualité
                 }
 
-                if (Items[i].SellIn < 0)
+                if (item.SellIn < 0)
                 {
-                    if (Items[i].Name != "Aged Brie")
+                    if (isNotAgedbrie)
                     {
-                        if (Items[i].Name != "Backstage passes to a TAFKAL80ETC concert")
+                        if (!SulfurasQualityCase(item))
```

Comment:
Tu appelles ici une méthode dont le nom m'indique qu'elle sert à vérifier si c'est le cas d'in produit Sulfuras.

Plus bas, je vois que Sulfuras fait en plus un traitement et modifie des variables => incohérence.